admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/componentexplorer/template.vue

```diff
@@ -23,161 +23,174 @@
-            </ul>
-    </div>
+  <div class="component-explorer">
+    <span class="panel-title">{{ $i18n('Components') }}</span>
+    <input type="text"
+           v-model="state.filter"
+           v-bind:placeholder="$i18n('filterComponents')"
+           tabindex="1"
+           autofocus/>
+    <select class="browser-default" v-model="state.group">
+      <option value="">{{ $i18n('allGroups') }}</option>
+      <option v-for="(group, key) in allGroups" v-bind:value="key" v-bind:key="key">{{
+          key
+        }}
+      </option>
+    </select>
+    <ul>
+      <li
+          v-for="(group, key) in groups" v-bind:key="key">
+        <div>
+          <ul class="collection">
+            <li
+                class="collection-item" style="cursor: move;"
+                v-for="component in group"
+                v-bind:key="componentKey(component)"
+                v-on:dragstart="onDragStart(component, $event)"
+                v-on:dragend="onDragEnd(component, $event)"
+                draggable="true">
+              <div>
+                <i class="material-icons">drag_handle</i>
+                <span>{{ displayName(component) }}</span>
+              </div>
+              <img v-if="component.thumbnail" v-bind:src="component.thumbnail">
+            </li>
+          </ul>
+        </div>
+      </li>
+    </ul>
+  </div>
 </template>
 
 <script>
-    import { IgnoreContainers } from '../../../../../../js/constants.js';
+import { IgnoreContainers } from '../../../../../../js/constants.js';
 
-    export default {
-        props: ['model'],
-        computed: {
-            state() {
-                const state = $perAdminApp.getNodeFromView('/state/componentExplorer')
-                if(state) {
-                    return state
-                }
-                Vue.set($perAdminApp.getView().state, 'componentExplorer', {filter: '', group: ''})
-                return $perAdminApp.getNodeFromView('/state/componentExplorer')
-            },
-            filteredList: function() {
-                var currentGroup = this.state.group
-                if (!this.$root.$data.admin.components) return {}
-                var componentPath = this.$root.$data.pageView.path.split('/')
-                var allowedComponents = ['/apps/' + componentPath[2]+ '/']
-                var list = this.$root.$data.admin.components.data
-                if (!list || !allowedComponents) return {}
 
-                var sorted = list.sort(function( left, right) {
-                    const leftName = (left.group + '-' + left.title).toLowerCase();
-                    const rightName = (right.group + '-' + right.title).toLowerCase();
-                    if(leftName < rightName) return -1;
-                    if(leftName > rightName) return 1;
-                    return 0;
-                })
+export default {
+  props: ['model'],
+  computed: {
+    state() {
+      const state = $perAdminApp.getNodeFromView('/state/componentExplorer');
```

Comment:
I see many semicolons `;` added in a couple of files here. I think we should get rid of them since the prevailing approach of the project is to drop them in `js`.

Reply:
sorry my bad. Wrong code-style.
We should in the near future try to have a style XML file that everyone can import into his IDE